src/overlays/actors/ovl_En_Door/z_en_door.c

```diff
@@ -44,36 +44,66 @@ const ActorInit En_Door_InitVars = {
     (ActorFunc)EnDoor_Draw,
 };
 
+typedef struct {
+    /* 0x00 */ s16 sceneNum;
+    /* 0x02 */ u8 dListIndex;
+    /* 0x04 */ s16 objectId;
+} EnDoorInfo;
+
+typedef enum {
+    /* 0 */ DOOR_DL_GAMEPLAY_KEEP,
+    /* 1 */ DOOR_DL_HIDAN,
+    /* 2 */ DOOR_DL_MIZU,
+    /* 3 */ DOOR_DL_HAKA,
+    /* 4 */ DOOR_DL_GAMEPLAY_FIELD_KEEP,
+    /* 5 */ DOOR_DL_MAX
+} EnDoorDListIndex;
+
 /**
  * Controls which object and display lists to use in a given scene
  */
 static EnDoorInfo sDoorInfo[] = {
-    { SCENE_HIDAN, 1, OBJECT_HIDAN_OBJECTS },
-    { SCENE_MIZUSIN, 2, OBJECT_MIZU_OBJECTS },
-    { SCENE_HAKADAN, 3, OBJECT_HAKA_DOOR },
-    { SCENE_HAKADANCH, 3, OBJECT_HAKA_DOOR },
+    { SCENE_HIDAN, DOOR_DL_HIDAN, OBJECT_HIDAN_OBJECTS },
+    { SCENE_MIZUSIN, DOOR_DL_MIZU, OBJECT_MIZU_OBJECTS },
+    { SCENE_HAKADAN, DOOR_DL_HAKA, OBJECT_HAKA_DOOR },
+    { SCENE_HAKADANCH, DOOR_DL_HAKA, OBJECT_HAKA_DOOR },
     // KEEP objects should remain last and in this order
-    { -1, 0, OBJECT_GAMEPLAY_KEEP },
-    { -1, 4, OBJECT_GAMEPLAY_FIELD_KEEP },
+    { -1, DOOR_DL_GAMEPLAY_KEEP, OBJECT_GAMEPLAY_KEEP },
+    { -1, DOOR_DL_GAMEPLAY_FIELD_KEEP, OBJECT_GAMEPLAY_FIELD_KEEP },
 };
 
 static InitChainEntry sInitChain[] = {
     ICHAIN_U8(targetMode, 0, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_STOP),
 };
 
-static AnimationHeader* sDoorAnims[] = { &gDoor3Anim, &gDoor1Anim, &gDoor4Anim, &gDoor2Anim };
+static AnimationHeader* sDoorAnims[KNOB_ANIM_MAX] = {
+    &gDoor3Anim, // KNOB_ANIM_ADULT_L
+    &gDoor1Anim, // KNOB_ANIM_CHILD_L
+    &gDoor4Anim, // KNOB_ANIM_ADULT_R
+    &gDoor2Anim, // KNOB_ANIM_CHILD_R
```

Comment:
Could you name these 4 animations now since you know what they are? Or are you saving that for a separate PR?

Reply:
I took the `knob_` enum from existing z_player documentation, which I think may have been made by just looking at z_player code. I haven't been able to see these anims for myself so I don't know if it checks out. I'll see what I can figure out